File: leetcode/python/0349_intersection_of_two_arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        set1 = { num1 for num1 in nums1 }

        res = set()
        for num2 in nums2:
            if num2 in set1: 
                res.add(num2)
        
        return list(res)

File: leetcode/java/0349_intersection_of_two_arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<Integer>();

        for (int i = 0; i < nums1.length; i++) {
            set1.add(nums1[i]);
        }

        int[] res = new int[nums2.length];
        int prevNum = nums2[0], j = 0;
        Arrays.sort(nums2);
        for (int i = 0; i < nums2.length; i++) {
            if (set1.contains(nums2[i]) && (i == 0 || prevNum != nums2[i])) {
                res[j++] = nums2[i];
            }

            if (prevNum != nums2[i]) prevNum = nums2[i];
        }

        return Arrays.copyOfRange(res, 0, j);
    }
}